exercises/perfect-numbers/perfect_numbers_test.py

```diff
@@ -1,42 +1,78 @@
+    def test_smallest_perfect_number(self):
+        self.assertIs(classify(6), "perfect")
 
-    def test_first_perfect_number(self):
-        self.assertIs(is_perfect(6), True)
+    def test_medium_perfect_number(self):
+        self.assertIs(classify(28), "perfect")
 
-    def test_no_perfect_number(self):
-        self.assertIs(is_perfect(8), False)
+    def test_large_perfect_number(self):
+        self.assertIs(classify(33550336), "perfect")
 
-    def test_second_perfect_number(self):
-        self.assertIs(is_perfect(28), True)
+    # Additional tests for this track
+    def test_third_perfect_number(self):
+        self.assertIs(classify(496), "perfect")
 
-    def test_abundant(self):
-        self.assertIs(is_perfect(20), False)
+    def test_fourth_perfect_number(self):
+        self.assertIs(classify(8128), "perfect")
 
-    def test_answer_to_the_ultimate_question_of_life(self):
-        self.assertIs(is_perfect(42), False)
+    def test_sixth_perfect_number(self):
+        self.assertIs(classify(8589869056), "perfect")
 
-    def test_third_perfect_number(self):
-        self.assertIs(is_perfect(496), True)
+    def test_seventh_perfect_number(self):
+        self.assertIs(classify(137438691328), "perfect")
+
+
+class AbundantNumbersTest(unittest.TestCase):
+    def test_smallest_abundant_number(self):
+        self.assertIs(classify(12), "abundant")
+
+    def test_medium_abundant_number(self):
+        self.assertIs(classify(30), "abundant")
+
+    def test_large_abundant_number(self):
+        self.assertIs(classify(33550335), "abundant")
+
+    # Additional tests for this track
+    def test_answer_to_the_ultimate_question_of_life(self):
+        self.assertIs(classify(42), "abundant")
 
     def test_odd_abundant(self):
-        self.assertIs(is_perfect(945), False)
+        self.assertIs(classify(945), "abundant")
 
-    def test_fourth_perfect_number(self):
-        self.assertIs(is_perfect(8128), True)
+    def test_even_abundant(self):
+        self.assertIs(classify(20), "abundant")
 
-    def test_fifth_perfect_number(self):
-        self.assertIs(is_perfect(33550336), True)
 
-    def test_sixth_perfect_number(self):
-        self.assertIs(is_perfect(8589869056), True)
+class DeficientNumbersTest(unittest.TestCase):
+    def test_smallest_prime_deficient_number(self):
+        self.assertIs(classify(2), "deficient")
 
-    def test_seventh_perfect_number(self):
-        self.assertIs(is_perfect(137438691328), True)
+    def test_smallest_nonprime_deficient_number(self):
+        self.assertIs(classify(4), "deficient")
+
+    def test_medium_deficient_number(self):
+        self.assertIs(classify(32), "deficient")
+
+    def test_large_deficient_number(self):
+        self.assertIs(classify(33550337), "deficient")
+
+    def test_edge_case(self):
+        self.assertIs(classify(1), "deficient")
+
+
+class InvalidInputsTest(unittest.TestCase):
+    def test_zero(self):
+        self.assertRaises(ValueError, classify, 0)
```

Comment:
By [convention](https://github.com/exercism/python/blob/78dfc6740fb962085d5f215e9b837e4292a20687/README.md#conventions) we use context manager syntax for `assertRaises`. Example:
```Python
with self.assertRaises(ValueError):
    classify(0)
```

Also, if the [canonical data](https://github.com/exercism/problem-specifications/blob/16f6dbcf4239099d25049a33f512ce936a14119c/exercises/perfect-numbers/canonical-data.json) specifies an error string, it is best practice to use [`assertRaisesRegexp`](https://docs.python.org/2/library/unittest.html#unittest.TestCase.assertRaisesRegexp) to check that the raised error contains the correct reason.